docs/cow-protocol/concepts/order-types/pay-debt-flash-loans.md

```diff
@@ -0,0 +1,104 @@
+
+#### Lender allows a third-party to repay your debt
+
+In this case, the user can sign a pre-hook that deploys a [cowshed](../../reference/sdks/cow-sdk/classes/CowShedHooks.md) to repay the debt using flash-loaned tokens. The underlying user order is then solely for repaying the required flash loan. Since the repayment is handled via `cowshed`, the user can be either an EOA (Externally Owned Account) or a contract (e.g., a SAFE wallet).
+
+#### Lender does not allow a third-party to repay your debt
+
+If the lender does not allow third-party debt repayment, the process becomes more involved. In this case, the repayment must come directly from the owner's account, meaning the owner cannot simply be an EOA (Externally Owned Account). Instead, it must be a smart contract (e.g.,a SAFE wallet), which can execute more complex transactions atomically. This is important because a contract can facilitate operations such as selling assets and repaying the debt in a single transaction, something an EOA cannot do.
+
+Additionally, repayment might not always involve selling the collateral token directly. In some cases, the protocol requires selling an interest-bearing version of the token instead. For example, if the collateral is aUSDC (which represents USDC deposited in Aave and earning interest), selling aUSDC instead of withdrawing and selling USDC directly ensures a more seamless repayment process. Understanding these trade-offs helps determine the correct approach based on the specific requirements of the lending protocol.
+
+##### Example
+
+Let's say user (SAFE with address `0x123...321`) borrowed 2000 USDC against 1 ETH of collateral in AAVE, and now wants to repay their debt position:
+1. The user first needs to determine the total repayment amount including accumulated interest: in this case, 2100 USDC is required to reclaim their 1 ETH collateral.
+2. To ensure complete debt repayment, the user should set the buy amount to 2101 USDC (adding a small buffer). This approach:
+   - Guarantees the debt is fully repaid
+   - May result in a small amount of unused USDC ("dust") returned to the user
+   - Prevents the scenario where the user ends up with remaining debt dust and must make an additional transaction
+3. The user needs to define two pre-hooks: one to repay the debt with collateral and the other for withdrawing the collateral repaid. In order to define both pre-hooks, the user needs to build both transactions with their SAFE:
+   - Repay the debt by calling the [repay](https://etherscan.io/address/0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2#writeProxyContract#F17) function of the AAVE's contract.
+   - Withdraw the collateral by calling the [withdraw](https://etherscan.io/address/0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2#writeProxyContract#F33) function of the AAVE's contract.
+4. The user places a buy order:
+
+```json
```

Comment:
Plain JSON doesn't support comments or trailing commas which leads to ugly rendering here. This is supported by JSON5. Not sure if this change would actually fix the rendering issue, though.
```suggestion
```json5
```